spec/controllers/collection_participants_controller_spec.rb

```diff
@@ -0,0 +1,464 @@
+      end
+    end
+  end
+
+  describe "index" do
+    let(:collection) { FactoryGirl.create(:collection) }
+    let(:current_role) { CollectionParticipant::NONE }
+    let(:user) { FactoryGirl.create(:user) }
+
+    context "user is not logged in" do
+      it "redirects to the index and displays an access denied message" do
+        get :index, collection_id: collection.name
+        it_redirects_to_with_error(new_user_session_path, "Sorry, you don't have permission to access the page you were trying to reach. Please log in.")
+      end
+    end
+
+    context "user is logged in" do
+      let!(:participant) do
+        FactoryGirl.create(
+          :collection_participant,
+          collection: collection,
+          pseud: user.default_pseud,
+          participant_role: current_role
+        )
+      end
+      let(:params) { { collection_id: collection.name } }
+
+      before do
+        fake_login_known_user(user)
+      end
+
+      context "where the user is not a maintainer" do
+        it "redirects to the index" do
+          get :index, params
+          it_redirects_to_with_error(user_path(user), "Sorry, you don't have permission to access the page you were trying to reach.")
+        end
+      end
+
+      context "where the user is a maintainer" do
+        render_views
+        let(:current_role) { CollectionParticipant::MODERATOR }
+
+        context "where the collection has several participants" do
+          let!(:users) do
+            Array.new(3) do
+              user = FactoryGirl.create(:user)
+              FactoryGirl.create(
+                :collection_participant,
+                collection: collection,
+                pseud: user.default_pseud
+              )
+              user
+            end
+          end
+
+          it "displays the participants in the correct order" do
+            get :index, params
+            expect(response).to have_http_status(:success)
+            users.each do |user|
+              expect(response.body).to include user.default_pseud.name
+            end
+          end
+        end
+      end
+    end
+  end
+
+  describe "update" do
+    let(:user) { FactoryGirl.create(:user) }
+    let(:collection) { FactoryGirl.create(:collection) }
+    let(:user_role) { CollectionParticipant::NONE }
+    let!(:user_participant) do
+      FactoryGirl.create(
+        :collection_participant,
+        pseud: user.default_pseud,
+        collection: collection,
+        participant_role: user_role
+      )
+    end
+    let(:id_to_update) { nil }
+    let(:params) do
+      {
+        collection_participant: {
+          participant_role:  CollectionParticipant::MEMBER,
+          id: id_to_update
+        }
+      }
+    end
+
+    before do
+      fake_login_known_user(user)
+    end
+
+    context "where there is no participant" do
+      it "displays an error and redirects to the index" do
+        put :update, params
+        expect(response).to have_http_status :redirect
```

Comment:
This might be a good place to use your `it_redirects_to_with_error` method (also applies to the other tests in this group that check for a redirect and flash error)